pinot-core/src/main/java/com/linkedin/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -203,6 +203,13 @@ public BrokerResponseNative reduceOnDataTable(@Nonnull BrokerRequest brokerReque
           setGroupByHavingResults(brokerResponseNative, aggregationFunctions, aggregationFunctionSelectStatus,
               brokerRequest.getGroupBy(), dataTableMap, brokerRequest.getHavingFilterQuery(),
               brokerRequest.getHavingFilterSubQueryMap());
+          if (brokerMetrics != null) {
+            long groupBySize = 0L;
+            for (AggregationResult aggregationResult : brokerResponseNative.getAggregationResults()) {
+              groupBySize = Math.max(aggregationResult.getGroupByResult().size(), groupBySize);
+            }
+            brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.GROUP_BY_SIZE, groupBySize);
```

Comment:
We should not emit the metric if the query does not perform `group by`. Current implementation seems to emit 0 for that case.

Reply:
It's inside the branch of group-by query, so should be good